src/server.rs

```diff
@@ -115,9 +115,11 @@ impl Service for Server {
         // from http://jaketrent.com/post/https-redirect-node-heroku/
         if let Some(raw) = req.headers().get_raw("x-forwarded-proto") {
             if raw != &b"https"[..] {
+                let host: &Host = req.headers().get().unwrap();
```

Comment:
This does work! It shouldn't be *strictly* needed, as I think you could do something like.....

    let host = req.headers().get().unwrap::<&Host>();

I *think*. I'm also not sure if this is better or worse looking, honestly 😄  Also, arguably

    let host = req.headers().get::<Option<&Host>>().unwrap();

What do you think?